src/Microsoft.DotNet.Build.Tasks/Targets/PerfTesting.targets

```diff
@@ -1,36 +1,59 @@
   </PropertyGroup>
 
-  <!-- perf events for consumptive metrics testing -->
-  <PropertyGroup Condition="'$(CollectConsumptiveMetrics)' == 'true'">
-    <ClrMetrics>GCSampledObjectAllocationHigh,GCSampledObjectAllocationLow</ClrMetrics>
-    <KernelMetrics>ImageLoad</KernelMetrics>
+  <!-- Perf Analysis NuGet package paths -->
+  <PropertyGroup>
+    <XunitPerfAnalysisPackageId>Microsoft.DotNet.xunit.performance.analysis</XunitPerfAnalysisPackageId>
+    <XunitPerfAnalysisPackageVersion>1.0.0-alpha-build0012</XunitPerfAnalysisPackageVersion>
+    <XunitPerfAnalysisPath>$(PackagesDir)$(XunitPerfAnalysisPackageId)/$(XunitPerfAnalysisPackageVersion)/tools/xunit.performance.analysis.exe</XunitPerfAnalysisPath>
   </PropertyGroup>
 
-  <!-- command line args to EventTracer.exe -->
+  <!-- Executable properties -->
   <PropertyGroup>
-    <EventTracerDataFile>$(TargetFileName).etl</EventTracerDataFile>
-    <EventTracerMetrics>-c $(ClrMetrics) -k $(KernelMetrics)</EventTracerMetrics>
-    <EventTracerReport>Perf.$(MSBuildProjectName).xml</EventTracerReport>
+    <XunitRunnerArgs>$(TargetFileName) -runnerhost $(TestHost) -runner $(XunitExecutable) -runid $(PerfRunId) -verbose</XunitRunnerArgs>
+    <PerfTestCommandLine>$(XunitPerfRunnerPath) $(XunitRunnerArgs) $(AdditionalPerfTestArgs)</PerfTestCommandLine>
+
+    <PerfRunOutputFile>$(TestPath)$(DebugTestFrameworkFolder)/$(PerfRunId).xml</PerfRunOutputFile>
+    <XunitAnalysisArgs>$(PerfRunOutputFile) -html $(AnalysisReportFileName) $(AdditionalXunitAnalysisArgs)</XunitAnalysisArgs>
+    <PerfAnalysisCommandLine>$(XunitPerfAnalysisPath) $(XunitAnalysisArgs)</PerfAnalysisCommandLine>
   </PropertyGroup>
 
+  <!-- Override test settings if RunPerfTestsForProject == true -->
+  <PropertyGroup Condition="'$(RunPerfTestsForProject)' == 'true'">
+    <TestCommandLine>$(PerfTestCommandLine)</TestCommandLine>
+  </PropertyGroup>
+
+  <!-- NOTE: Temporary workaround, there is some issue where the discoverer requires this to be in the target directory -->
+  <Target Name="CopyXunitExecutionDesktop" BeforeTargets="RunTestsForProject"
+          Condition="'$(RunPerfTestsForProject)' == 'true'">
+    <Copy SourceFiles="$(XunitPerfRunnerPackageDir)/tools/xunit.execution.desktop.dll"
+          DestinationFolder="$(TestPath)$(DebugTestFrameworkFolder)"
+          SkipUnchangedFiles="true" />
+  </Target>
+
+  <Target Name="AnalyzePerfResults"
+          AfterTargets="RunTestsForProject"
+          Condition="'$(AnalyzePerfResults)' == 'true' and Exists('$(PerfRunOutputFile)')">
+    <Exec Command="$(PerfAnalysisCommandLine)"
+          WorkingDirectory="$(TestPath)%(TestTargetFramework.Folder)"
+          ContinueOnError="false">
+      <Output PropertyName="PerfTestRunExitCode" TaskParameter="ExitCode" />
+    </Exec>
+    <Message Text="Performance test analysis report is available at $(TestPath)$(DebugTestFrameworkFolder)/$(AnalysisReportFileName)" />
```

Comment:
This property is a bit weirdly named, but it's the right one. "TestPath" actually includes most of the path, i.e. ````bin\tests\Windows_NT.AnyCPU.Debug\<projectname> ```` whereas "DebugTestFrameworkFolder" is just ````dnxcore50```` or ````net45````